dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/jaxb/JAXBContextCache.java

```diff
@@ -0,0 +1,504 @@
+                public JAXBContext run() throws Exception {
+                    return JAXBContext.newInstance(classes.toArray(new Class<?>[0]), map);
+                }
+            });
+        } catch (PrivilegedActionException e2) {
+            if (e2.getException() instanceof JAXBException) {
+                JAXBException ex = (JAXBException)e2.getException();
+                if (map.containsKey("com.sun.xml.bind.defaultNamespaceRemap")
+                    && ex.getMessage() != null
+                    && ex.getMessage().contains("com.sun.xml.bind.defaultNamespaceRemap")) {
+                    map.put("com.sun.xml.internal.bind.defaultNamespaceRemap",
+                            map.remove("com.sun.xml.bind.defaultNamespaceRemap"));
+                    ctx = JAXBContext.newInstance(classes.toArray(new Class<?>[0]), map);
+                } else {
+                    throw ex;
+                }
+            } else {
+                throw new RuntimeException(e2.getException());
+            }
+        }
+        return ctx;
+    }
+    // Now we can not add all the classes that Jaxb needed into JaxbContext,
+    // especially when
+    // an ObjectFactory is pointed to by an jaxb @XmlElementDecl annotation
+    // added this workaround method to load the jaxb needed ObjectFactory class
+    private static boolean addJaxbObjectFactory(JAXBException e1, Set<Class<?>> classes) {
+        boolean added = false;
+        java.io.ByteArrayOutputStream bout = new java.io.ByteArrayOutputStream();
+        java.io.PrintStream pout = new java.io.PrintStream(bout);
+        e1.printStackTrace(pout);
+        String str = new String(bout.toByteArray());
+        Pattern pattern = Pattern.compile("(?<=There's\\sno\\sObjectFactory\\swith\\san\\s"
+                                          + "@XmlElementDecl\\sfor\\sthe\\selement\\s\\{)\\S*(?=\\})");
+        java.util.regex.Matcher matcher = pattern.matcher(str);
+        while (matcher.find()) {
+            String pkgName = JAXBUtils.namespaceURIToPackage(matcher.group());
+            try {
+                Class<?> clz = JAXBContextCache.class.getClassLoader()
+                    .loadClass(pkgName + "." + "ObjectFactory");
+
+                if (!classes.contains(clz)) {
+                    classes.add(clz);
+                    added = true;
+                }
+            } catch (ClassNotFoundException e) {
+                // do nothing
+            }
+
+        }
+        return added;
+    }
+
+    public static void addPackage(Set<Class<?>>  classes, String pkg, ClassLoader loader) {
+
+        boolean isLoggableInfo = LOG.isLoggable(Level.INFO);   // Liberty Change
+
+        try {
+            classes.add(Class.forName(pkg + ".ObjectFactory", false, loader));
+        } catch (Exception ex) {
+            //ignore
+	    // Liberty Change begin
+	    if (isLoggableInfo) {
+	       LOG.info("addPackage: Ignoring exception: " + ex);
```

Comment:
should this be trace?

Reply:
same comment as above.

Reply:
yes, will change all these to use fine or finest level